selection_tools/prepare_histograms.C

```diff
@@ -50,39 +50,72 @@ int main(int argc, char *argv[])
+    {"muon_theta",  new TH1D("muon_theta", "muon_theta", 1000, -100, 100)},
+    {"muon_lxy",  new TH1D("muon_lxy", "muon_lxy", 1000, 0, 100)},
+    {"muon_lz",  new TH1D("muon_lz", "muon_lz", 1000, 0, 100)},
+    {"muon_lxyz",  new TH1D("muon_lxyz", "muon_lxyz", 1000, 0, 100)},
+    {"muon_ctau",  new TH1D("muon_ctau", "muon_ctau", 1000, 0, 100)},
+    {"muon_boost",  new TH1D("muon_boost", "muon_boost", 1000, 0, 1000)},
+
+    {"muon_deltaR",  new TH1D("muon_deltaR", "muon_deltaR", 1000, 0, 100)},
+    {"muon_deltaPhi",  new TH1D("muon_deltaPhi", "muon_deltaPhi", 1000, -100, 100)},
     
     {"dimuon_pt",  new TH1D("dimuon_pt", "dimuon_pt", 1000, 0, 1000)},
     {"dimuon_pz",  new TH1D("dimuon_pz", "dimuon_pz", 1000, -1000, 1000)},
     {"dimuon_mass",  new TH1D("dimuon_mass", "dimuon_mass", 1000, 0, 100)},
+    {"dimuon_eta",  new TH1D("dimuon_eta", "dimuon_eta", 1000, -100, 100)},
+    {"dimuon_phi",  new TH1D("dimuon_phi", "dimuon_phi", 1000, -100, 100)},
     
     {"first_mother_pt",  new TH1D("first_mother_pt", "first_mother_pt", 1000, 0, 1000)},
     {"first_mother_pz",  new TH1D("first_mother_pz", "first_mother_pz", 1000, -1000, 1000)},
     {"first_mother_mass",  new TH1D("first_mother_mass", "first_mother_mass", 1000, 0, 200)},
+    {"first_mother_eta",  new TH1D("first_mother_eta", "first_mother_eta", 1000, -100, 100)},
+    {"first_mother_phi",  new TH1D("first_mother_phi", "first_mother_phi", 1000, -100, 100)},
     {"first_mother_lxy",  new TH1D("first_mother_lxy", "first_mother_lxy", 1000, 0, 100)},
     {"first_mother_lz",  new TH1D("first_mother_lz", "first_mother_lz", 1000, 0, 100)},
     {"first_mother_lxyz",  new TH1D("first_mother_lxyz", "first_mother_lxyz", 1000, 0, 100)},
     {"first_mother_ctau",  new TH1D("first_mother_ctau", "first_mother_ctau", 1000, 0, 100)},
     {"first_mother_boost",  new TH1D("first_mother_boost", "first_mother_boost", 1000, 0, 100)},
-    // TODO: add eta and delta R...
   };
   
   
   auto fill_single_muon_hists = [&](const Particle *muon)->void {
     hists["muon_pt"]->Fill(muon->four_vector.Pt());
     hists["muon_pz"]->Fill(muon->four_vector.Pz());
     hists["muon_mass"]->Fill(muon->four_vector.M());
+    hists["muon_eta"]->Fill(muon->four_vector.Eta());
+    hists["muon_phi"]->Fill(muon->four_vector.Phi());
+    hists["muon_y"]->Fill(muon->four_vector.Rapidity());
+    hists["muon_theta"]->Fill(muon->four_vector.Theta());
+    hists["muon_lxy"]->Fill(sqrt(pow(muon->x, 2) + pow(muon->y, 2)));
+    hists["muon_lz"]->Fill(muon->z);
+    hists["muon_lxyz"]->Fill(sqrt(pow(muon->x, 2) + pow(muon->y, 2) + pow(muon->z, 2)));
+    hists["muon_ctau"]->Fill(muon->ctau);
+    hists["muon_boost"]->Fill(muon->momentum()/muon->mass);
+  };
+
+  auto fill_double_muon_hists =  [&](const Particle *muon1,const Particle *muon2)->void {
+    hists["muon_deltaPhi"]->Fill(muon1->four_vector.DeltaPhi(muon2->four_vector));
+    hists["muon_deltaR"]->Fill(muon1->four_vector.DeltaR(muon2->four_vector));
```

Comment:
Small thing - I think these two histograms should be called `dimuon_deltaPhi` and `dimuon_deltaR`, given there's no delta for just one muon ;)